process/block/bootstrapStorage/bootstrapStorer.go

```diff
@@ -0,0 +1,151 @@
+package bootstrapStorage
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+// HighestRoundFromBootStorage is the key for the highest round that is saved in storage
+const highestRoundFromBootStorage = "highestRoundFromBootStorage"
+
+// ErrNilMarshalizer signals that an operation has been attempted to or with a nil Marshalizer implementation
+var ErrNilMarshalizer = errors.New("nil Marshalizer")
+
+// ErrNilBootStorer signals that an operation has been attempted to or with a nil storer implementation
+var ErrNilBootStorer = errors.New("nil boot storer")
+
+//BootstrapHeaderInfo is struct used to store information about a header
+type BootstrapHeaderInfo struct {
+	ShardId uint32
+	Nonce   uint64
+	Hash    []byte
+}
+
+// BootstrapData is struct used to store information that are needed for bootstrap
+type BootstrapData struct {
+	HeaderInfo           BootstrapHeaderInfo
+	LastNotarizedHeaders []BootstrapHeaderInfo
+	LastFinals           []BootstrapHeaderInfo
+	ProcessedMiniBlocks  map[string]map[string]struct{}
+	HighestFinalNonce    uint64
+	LastRound            int64
+}
+
+type bootstrapStorer struct {
+	store       storage.Storer
+	marshalizer marshal.Marshalizer
+	lastRound   int64
+}
+
+// NewBootstrapStorer will return an instance of bootstrap storer
+func NewBootstrapStorer(
+	marshalizer marshal.Marshalizer,
+	store storage.Storer,
+) (*bootstrapStorer, error) {
+	if check.IfNil(marshalizer) {
+		return nil, ErrNilMarshalizer
+	}
+	if check.IfNil(store) {
+		return nil, ErrNilBootStorer
+	}
+
+	bootStorer := &bootstrapStorer{
+		store:       store,
+		marshalizer: marshalizer,
+	}
+	bootStorer.lastRound = bootStorer.GetHighestRound()
+
+	return bootStorer, nil
+}
+
+// Put will save bootData in storage
+func (bs *bootstrapStorer) Put(round int64, bootData BootstrapData) error {
+	bootData.LastRound = bs.lastRound
```

Comment:
this member is set inside Put method because we use last round as key for previous element from storage and only  bootstrapStorer know which was the last element.

Reply:
It could expose lastRound, but ok